### PR TITLE
fix and document warnings policy

### PR DESCRIPTION
**Pull request by thedavidmeister**
## changes

- fix all the warnings that i was seeing on local
- DENY WARNINGS!!!
- document denial of warnings as per Montreal hodl podl discussions
- document makefile/nix split

### CODE REVIEW

**Comment by zippy** [comment]
I note that I do still see this warning:

```
warning: redundant linker flag specified for library `zmq`  
```
while building this branch...

**Comment by thedavidmeister** [comment]
@zippy fwiw i've seen that in other branches too, i don't think it is new

File: README.md [hunk 1]
```diff
@@ -103,6 +115,57 @@ Run:
...
+
+The change may not be immediately apparent to you. The change may break the development environment on a different operating system, e.g. Windows.
+
+At the same time, we do not want to catastrophise and stifle innovation or legitimate upgrades.
+
+If you have a proposal to improve our CI config, that is great!
+
+Please open a dedicated branch for the change in isolation so we can discuss the proposal together.
+
+Please broadcase the proposal in chat to maximise visibility and the opportunity for everyone to respond.
```
Comment (lucksus) [inline README.md]: "broadcast" ?

Comment (lucksus) [inline README.md]:
```suggestion
Please broadcast the proposal in chat to maximise visibility and the opportunity for everyone to respond.
```

**Pull request merged**